app/models/grda_warehouse/auth_policies/policy_provider.rb

```diff
@@ -0,0 +1,127 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+require 'memery'
+
+class GrdaWarehouse::AuthPolicies::PolicyProvider
+  include Memery
+  attr_reader :user
+
+  # TODO: START_ACL remove after ACL migration is complete
+  ###
+  # The legacy permission policy only checks the permissions on a user's roles and assumes the loaded record is visible
+  # to the user already. This is a dangerous assumption so it's required to confirm the record is authorized by
+  # setting legacy_implicitly_assume_authorized_access to true.
+  # Note this is only relevant for users where using_acls is false
+  ###
+  attr_accessor :legacy_implicitly_assume_authorized_access
+  # END_ACL
+
+  def initialize(user)
+    @user = user
+  end
+
+  def for_client(client_or_id)
+    if user.using_acls?
+      client_id = id_from_arg(client_or_id, GrdaWarehouse::Hud::Client)
+      for_client_using_acls(client_id)
+    else
+      # TODO: START_ACL remove after ACL migration is complete
+      legacy_user_role_policy
+      # END_ACL
+    end
+  end
+
+  def for_patient(patient)
+    for_client(patient.client_id)
+  end
+
+  def for_project(project_or_id)
+    if user.using_acls?
+      project_id = id_from_arg(project_or_id, GrdaWarehouse::Hud::Project)
+      for_project_using_acls(project_id)
+    else
+      # TODO: START_ACL remove after ACL migration is complete
+      legacy_user_role_policy
+      # END_ACL
+    end
+  end
+
+  protected
+
+  memoize def for_project_using_acls(project_id)
+    p_t = GrdaWarehouse::Hud::Project.arel_table
+    collection_ids = GrdaWarehouse::ProjectCollectionMember.where(project_id: project_id).pluck(:collection_id)
```

Comment:
This seems to be assuming the project would be a member of the collection directly.  I'm probably missing it, but where do we check if the project is included in the collection if its organization or data source is included? (CoC Code is covered below.)  If this is already handled, great!  If not, I wonder if we should make a class method or scope on `Collection` that takes a project ID and returns the collections that contain it, just so we can collect up the logic for that in the place where the assignment takes place.

Reply:
The `ProjectCollectionMember` view should be handling that, it collections related through associated entities (joins project to orgs, data source, etc). I added test cases to prove it out in `policy_provider_spec.rb` - do those tests describe the desired behavior?

Reply:
Right!  I forgot you'd encoded that in the view, perfect solution.